hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -1239,9 +1236,8 @@ void startDataNode(Configuration conf,
     synchronized (this) {
       this.dataDirs = dataDirs;
     }
-    this.conf = conf;
-    this.dnConf = new DNConf(conf);
-    checkSecureConfig(dnConf, conf, resources);
+    this.dnConf = new DNConf(this);
```

Comment:
If we revert changes to DNConf we can just replace this with `this.dnConf = new DNConf(getConf())`.